wire-runtime/src/main/java/com/squareup/wire/RuntimeMessageAdapter.java

```diff
@@ -167,7 +167,10 @@ B newBuilder() {
       FieldBinding<M, B> fieldBinding = fieldBindings.get(tag);
       try {
         if (fieldBinding != null) {
-          Object value = fieldBinding.singleAdapter().decode(reader);
+          ProtoAdapter<?> adapter = fieldBinding.isMap()
+              ? fieldBinding.adapter()
```

Comment:
In this case `singleAdapter()` delegates to call `ProtoAdatper.get` of `WireField.adapter` which in `Map`'s case is the adapter for the value type only.

I can look to make it more logical. The runtime adapter and field binding classes are a bit of a mess right now.